### DISCUSSION TITLE
Add a template hook for action buttons on entry edit page in control panel

### DISCUSSION THREAD

**Discussion by ivanBereznev** [craftcms/cms#10420] [Ideas] [answered]
Currently there seems to be no clean way to add custom controls to action buttons on entry edit page:
![craft_action_buttons](https://user-images.githubusercontent.com/18012861/151350718-4bd0296d-b5f7-42a5-8df4-3b9845101a25.png)
The only way to add custom controls there is jQuery hacks including URL matching in case you want to show controls depending on entry type.

It would be nice to have a clean way to add custom controls there, I think template hooks is approach used in other places, so probably one fits here as well. And it would be very nice to have the entry being rendered available in context provided to hook functions, it will make it easy to customize controls based on a particular entry type.

**Comment by brandonkelly** [comment, marked as answer]
Just added a new `EVENT_DEFINE_ADDITIONAL_BUTTONS` action to `craft\base\Element` which can be used to add additional buttons there, for Craft 4. (6a189f90).

**Reply by stevecomrie** [reply]
For anyone else that finds their way here with a similar question, as of 4.3, this event is now called:

`EVENT_DEFINE_ADDITIONAL_BUTTONS`

per: https://github.com/craftcms/cms/discussions/10741

**Reply by brandonkelly** [reply]
Thanks for the reminder! Just updated my last comment.
